pkg/rhelv2/rpm/rpm.go

```diff
@@ -91,34 +186,26 @@ func listFeatures(files tarutil.LayerFiles, queryFmt string) ([]*database.RHELv2
 		return nil, nil, err
 	}
 
-	f, hasFile := files.Get(dbPath)
-	if !hasFile {
-		return nil, cpes, nil
-	}
-
 	defer metrics.ObserveListFeaturesTime(pkgFmt, "all", time.Now())
 
-	// Write the required "Packages" file to disk
-	tmpDir, err := os.MkdirTemp("", "rpm")
+	rpmDB, err := createRPMDatabaseFromImage(files)
 	if err != nil {
-		log.WithError(err).Error("could not create temporary folder for RPM detection")
-		return nil, nil, commonerr.ErrFilesystem
+		return nil, nil, err
 	}
-	defer func() {
-		_ = os.RemoveAll(tmpDir)
-	}()
-
-	err = os.WriteFile(tmpDir+"/Packages", f.Contents, 0700)
-	if err != nil {
-		log.WithError(err).Error("could not create temporary file for RPM detection")
-		return nil, nil, commonerr.ErrFilesystem
+	if rpmDB == nil {
+		return nil, cpes, nil
 	}
 
+	defer rpmDB.delete()
 	defer metrics.ObserveListFeaturesTime(pkgFmt, "cli+parse", time.Now())
 
-	cmd := exec.Command("rpm",
-		`--dbpath`, tmpDir,
-		`--query`, `--all`, `--queryformat`, queryFmt)
+	cmd := exec.Command(
```

Comment:
> What do we do from there? 

We let `rpm` do its thing, which is using `.sqIite` (the new model) and default to BDB if the new model is unavailable. I am assuming that's the expected behavior in the scanned image itself since it should be using a compatible rpm version (likely). But it's possible the rpm version in the image is configured to ignore either type of DB, just FYI.